### ISSUE TITLE
Non-object is blowing up function-call

### ISSUE THREAD

**Issue by nuflowx** [dispix/deep-diff#3]
So when I pass in nested objects that include various different children, I was getting an error `First parameter must be an object`.

After debugging I found that even though the key is `undefined` (not sure what type the key actually was) if continued in the `if` clause:

`if (typeof obj2[key] === 'object')`
https://github.com/dispix/deep-diff/blob/master/src/deepDiff.js#L40

I used lodash `isPlainObject`-function and this did notice it was not actually a function.

`typeof` is not very reliable check as everything (or a lot of things) in javascript are considered objects. So an improvement might be needed in that check.

*This only happened with more complex objects and I ran into this problem after using deepDiff with more complexly nested Objects*

I might be able to open a PR in a few days, not sure when I have time to spare next. For now I just copied the code and use it straight in my application instead of including it as a package

**Comment by dispix** [comment]
Thanks for the feedback ! I'll probably have the time to take a look during the week but if you feel like opening a PR you are most welcome to do so :)

**Comment by nuflowx** [comment]
During the week it's less likely for me. I'll keep an eye on it for the weekend in case you don't get 'round to it until then ;)

**Comment by dispix** [comment]
Could you please give me a scenario to reproduce the bug you encountered ?

**Comment by nuflowx** [comment]
I'll get back to you later today, busy atm.

**Comment by dispix** [comment]
@nuflowx any news on that scenario ? I can't find a use case where I encounter what you describe :/

**Comment by dmnrmr** [comment]
```
deepDiff({}, {
  foo: {
    bar: 'baz'
  }
}, true);
```

**Comment by dispix** [comment]
@dmnrmr Thanks for your comment. I fixed this and shipped it in `v0.2.8`.

**Comment by dmnrmr** [comment]
I am pretty sure that if one of properties is a function it will also fail. Will test new version in a few hours.

**Comment by dmnrmr** [comment]
```
deepDiff(
      {
        mapConfig: {
          fullscreenControl: true,
          showPropertyLayer: true
        },
        geojson: {
          type: 'FeatureCollection',
          features: [
            {
              type: 'Feature',
              geometry: {
                type: 'Point',
                coordinates: [
                  267193.58,
                  6641973.82
                ]
              },
              crs: {
                type: 'name',
                properties: {
                  name: 'urn:ogc:def:crs:EPSG::25833'
                }
              },
              properties: {
                id: '301-11167-10',
                title: 'Dalssvingen 10',
                icon: 'thumb-tack',
                type: 'group'
              }
            }
          ],
          crs: {
            type: 'name',
            properties: {
              name: 'urn:ogc:def:crs:EPSG::25833'
            }
          }
        },
        onFeatureClick: () => null
      } , {
        geojson: {
          type: 'Feature',
          geometry: {
            type: 'Point',
            coordinates: [
              267193.58,
              6641973.82
            ]
          },
          crs: {
            type: 'name',
            properties: {
              name: 'urn:ogc:def:crs:EPSG::25833'
            }
          },
          properties: {
            id: '301-11167-10',
            title: 'Dalssvingen 10',
            icon: 'thumb-tack',
            type: 'group',
            cadastreKey: '0301/169/7/0/0'
          }
        }
      }, true
    ));
```

Also fails.

**Comment by dispix** [comment]
@dmnrmr This test passes correctly with the `v0.2.8` on my side.

**Comment by dispix** [comment]
@dmnrmr Found out why you still have an issue with that case: the `v0.2.8` wasn't including the new build so you were still really in `v0.2.7` (so that's totally my bad).

I've updated the scripts so this kind of issue doesn't happen again and released a new `v0.2.9` version that includes the changes.

**Comment by dmnrmr** [comment]
Works fine now. Thank you.
